### PR TITLE
.coafile: Use RSTcheckBear and update .coafile

### PR DESCRIPTION
**Pull request by mridubhatnagar**
Update .coafile with RSTcheckBear.

Closes https://github.com/coala/coala/issues/4209

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by mridubhatnagar** [comment]
Three errors are still coming which can be resolved by creating seperate issues for each. 
Other fixes I have made.

**Comment by mridubhatnagar** [comment]
For possible underline too short. I made changes but it is still not fixed.

**Comment by mridubhatnagar** [comment]
Updated languagetools and docs. For all rst docs.

**Comment by Vamshi99** [comment]
Also, update invalidlinks section. @mridubhatnagar

**Comment by mridubhatnagar** [comment]
Updated.

**Comment by mridubhatnagar** [comment]
For the mentioned issues are we opening new issues? 
Could not fix them hence had left.

**Comment by mridubhatnagar** [comment]
@Vamshi99 can we create issues for the above mentioned 3 errors that need to be fixed? Rest all the needed changes are done.

File: .coafile [hunk 1]
```diff
@@ -40,7 +40,7 @@ ignore_length_regex = ^.*https?://
 [DOCS]
 bears = SpaceConsistencyBear, LineLengthBear
 
-files = docs/**/*.rst, CONTRIBUTING.rst, README.rst
+files = **.rst
```
Comment (Vamshi99) [inline .coafile]: This change makes bears not to check anything in docs/**/*.rst and similar in all cases.

Comment (mridubhatnagar) [inline .coafile]: shall check again. And revert.

Comment (mridubhatnagar) [inline .coafile]: On command prompt I re-checked.
Python 3.4.2 (v3.4.2:ab2c023a9432, Oct  6 2014, 22:15:05) [MSC v.1600 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from coalib.parsing.Globbing import glob
>>> glob("**.rst")
['CONTRIBUTING.rst', 'docs\\Developers\\Adding_CI.rst', 'docs\\Developers\\Bear_Installation_Tool.rst', 'docs\\Developers\\coala_settings.rst', 'docs\\Developers\\Codestyle.rst', 'docs\\Developers\\Development_Setup.rst', 'docs\\Developers\\Executing_Tests.rst', 'docs\\Developers\\External_Bears.rst', 'docs\\Developers\\Git_Basics.rst', 'docs\\Developers\\Linter_Bears_Advanced.rst', 'docs\\Developers\\Newcomers_Guide.rst', 'docs\\Developers\\Review.rst', 'docs\\Developers\\Testing_Bears.rst', 'docs\\Developers\\Useful_Links.rst', 'docs\\Developers\\Writing_Documentation.rst', 'docs\\Developers\\Writing_Good_Commits.rst', 'docs\\Developers\\Writing_Linter_Bears.rst', 'docs\\Developers\\Writing_Native_Bears.rst', 'docs\\Developers\\Writing_Tests.rst', 'docs\\index.rst', 'README.rst', 'RELEASE_NOTES.rst']

**Comment by mridubhatnagar** [comment]
On running coala on my local machine. Only one error of argparse came. Hence I made a commit

**Comment by mridubhatnagar** [comment]
@vam

File: .coafile (hunk 1 above)
Comment (mridubhatnagar) [inline .coafile]: @Vamshi99 can you please have a look at this one. And suggest me how to proceed furthur with the issues that are coming.

Comment (jayvdb) [inline .coafile]: This is causing it to also find `venv/local/lib/python2.7/site-packages/setuptools-28.8.0.dist-info/DESCRIPTION.rst` in https://circleci.com/gh/coala/coala/12557

Please use `docs/**/*.rst, *.rst` for all the rst sections.

**Comment by jayvdb** [comment]
This is blocked by

- [ ] https://github.com/coala/coala/issues/4221
- [ ] https://github.com/coala/coala/issues/4222

Also, the error occurring in `coala_settings.rst` is because rstcheck doesnt understand `argparse` which was added in https://github.com/coala/coala/pull/3315  .  You need to configure rstcheck to ignore `argparse`.

File: .coafile (hunk 1 above)
Comment (mridubhatnagar) [inline .coafile]: @jayvdb  thanks for looking into this. Will update soon.

**Comment by mridubhatnagar** [comment]
@jayvdb updated with the changes requested by you.

**Comment by mridubhatnagar** [comment]
@jayvdb updated the changes as requested by you.

**Comment by mridubhatnagar** [comment]
Locally when I run coala I am not getting the mentioned Git_Basics.rst error. 
I have resolved the _dev_notes error in the same PR itself. 
Reason:
To reproduce the mentioned errors the assignee will have to
make changes in the .coafile as well. Because in the current .coafile RSTcheckBear is not present.

Hence fixed it on my own.

**Comment by jayvdb** [comment]
> should it be ignore= docs/devlopers/coala_settings.rst so that it does not get to argparse.

That is a temporary workaround.

The correct fix will need an enhancement to RSTcheckBear : https://github.com/coala/coala-bears/issues/1723 - maybe you want to try that one? You have done enough 'newcomer' issues, and https://github.com/coala/coala-bears/issues/1723 is a fairly simple 'low' bug.

**Comment by mridubhatnagar** [comment]
When I took up this issue no issue label was assigned . When I was done with this it got labelled as newcomer. Yes sure will try out :)

**Comment by jayvdb** [comment]
There are other issues that need to be fixed first, before this can be completed.

**Comment by mridubhatnagar** [comment]
@shreyans800755 yes those changes were after adding rst checks in .coafile. 
Yes I am aware of _dev-notes part. Have to undo my last commit. My last commit has resolved both #4221 and #4222. 
Shall do it.

**Comment by jayvdb** [comment]
Force pushing this to get it finished.

File: coalib/bears/Bear.py [hunk 2]
```diff
@@ -223,6 +223,7 @@ def maintainers_emails(cls):
         """
         :return: A set containing ``MAINTAINERS_EMAILS`` if specified, else
                  takes ``AUTHORS_EMAILS`` by default.
+
```
Comment (jayvdb) [inline coalib/bears/Bear.py]: eh what is this doing in here.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/4211/commits/1941f5e0d1a2f10316e4d5328b691a146465c6e4 https://github.com/coala/coala/pull/4211/commits/f2b062535db89b3480758b5dc0b2b93b03c9c9e1

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**